parsers/keyval/keyval.go

```diff
@@ -0,0 +1,216 @@
+// Package keyval parses logs whose format is many key=val pairs
+package keyval
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/kr/logfmt"
+
+	"github.com/honeycombio/honeytail/event"
+	"github.com/honeycombio/honeytail/parsers"
+)
+
+var possibleTimeFieldNames = []string{
+	"time", "Time",
+	"timestamp", "Timestamp", "TimeStamp",
+	"date", "Date",
+	"datetime", "Datetime", "DateTime",
+}
+
+type Options struct {
+	TimeFieldName string `long:"timefield" description:"Name of the field that contains a timestamp"`
+	Format        string `long:"format" description:"Format of the timestamp found in timefield (supports strftime and Golang time formats)"`
+}
+
+type Parser struct {
+	conf       Options
+	lineParser LineParser
+	nower      Nower
+
+	warnedAboutTime bool
+}
+
+type Nower interface {
+	Now() time.Time
+}
+
+type RealNower struct{}
+
+func (r *RealNower) Now() time.Time {
+	return time.Now().UTC()
+}
+
+func (p *Parser) Init(options interface{}) error {
+	p.conf = *options.(*Options)
+
+	p.nower = &RealNower{}
+	p.lineParser = &KeyValLineParser{}
+	return nil
+}
+
+type LineParser interface {
```

Comment:
ehh, I'd rather not. The Parser interface is the slimmest connection - having a LineParser interface within the parser is actually not required at all.  Honestly I'm not sure why it's even here; there's only ever one implementation of the interface. (the MySQL parser, for example, doesn't have a LineParser anywhere.)